### PR TITLE
Added djinn war requirement back

### PR DESCRIPTION
**Pull request by newacc12**
Added the requirement to complete the missions in order to trade with the djinns.

### CODE REVIEW

File: data/npc/scripts/Alesar.lua [hunk 1]
```diff
@@ -79,10 +79,10 @@ local function creatureSayCallback(cid, type, msg)
 end
 
 local function onTradeRequest(cid)
---!--	if Player(cid):getStorageValue(Storage.DjinnWar.EfreetFaction.Mission03) ~= 3 then
---!--		npcHandler:say('I\'m sorry, but you don\'t have Malor\'s permission to trade with me.', cid)
---!--		return false
---!--	end
+	if Player(cid):getStorageValue(Storage.DjinnWar.EfreetFaction.Mission03) ~= 3 then
```
Comment (vankk) [inline data/npc/scripts/Alesar.lua]: Please add the `Player(cid)` to a variable and uses after.

File: data/npc/scripts/Haroun.lua [hunk 2]
```diff
@@ -51,10 +51,10 @@ local function creatureSayCallback(cid, type, msg)
 end
 
 local function onTradeRequest(cid)
---!--	if Player(cid):getStorageValue(Storage.DjinnWar.MaridFaction.Mission03) ~= 3 then
---!--		npcHandler:say('I\'m sorry, human. But you need Gabel\'s permission to trade with me.', cid)
---!--		return false
---!--	end
+	if Player(cid):getStorageValue(Storage.DjinnWar.MaridFaction.Mission03) ~= 3 then
```
Comment (vankk) [inline data/npc/scripts/Haroun.lua]: Same here.

File: data/npc/scripts/Nah'Bob.lua [hunk 3]
```diff
@@ -48,10 +48,10 @@ local function creatureSayCallback(cid, type, msg)
 end
 
 local function onTradeRequest(cid)
---!--	if Player(cid):getStorageValue(Storage.DjinnWar.MaridFaction.Mission03) ~= 3 then
---!--		npcHandler:say('I\'m sorry, human. But you need Gabel\'s permission to trade with me.', cid)
---!--		return false
---!--	end
+	if Player(cid):getStorageValue(Storage.DjinnWar.MaridFaction.Mission03) ~= 3 then
```
Comment (vankk) [inline data/npc/scripts/Nah'Bob.lua]: Same here.

File: data/npc/scripts/Yaman.lua [hunk 4]
```diff
@@ -81,10 +81,10 @@ local function creatureSayCallback(cid, type, msg)
 end
 
 local function onTradeRequest(cid)
---!--	if Player(cid):getStorageValue(Storage.DjinnWar.EfreetFaction.Mission03) ~= 3 then
---!--		npcHandler:say('I\'m sorry, but you don\'t have Malor\'s permission to trade with me.', cid)
---!--		return false
---!--	end
+	if Player(cid):getStorageValue(Storage.DjinnWar.EfreetFaction.Mission03) ~= 3 then
```
Comment (vankk) [inline data/npc/scripts/Yaman.lua]: Same here.

**Review by vankk — COMMENTED** [review]
Also, could have a variable option on global.lua if we wanna the enable Djinn Missions or not.

File: data/npc/scripts/Alesar.lua (hunk 1 above)
Comment (newacc12) [reply data/npc/scripts/Alesar.lua]: Done. Is there a specific reason for this other than readability?

File: data/npc/scripts/Haroun.lua (hunk 2 above)
Comment (newacc12) [reply data/npc/scripts/Haroun.lua]: Done

File: data/npc/scripts/Nah'Bob.lua (hunk 3 above)
Comment (newacc12) [reply data/npc/scripts/Nah'Bob.lua]: Done

File: data/npc/scripts/Yaman.lua (hunk 4 above)
Comment (newacc12) [reply data/npc/scripts/Yaman.lua]: Done

**Comment by newacc12** [comment]
> Also, could have a variable option on global.lua if we wanna the enable Djinn Missions or not.

We could. I just modified based on tibia global as per https://github.com/opentibiabr/OTServBR-Global/issues/463. 

I would also be more inclined to have a more generic variable like requireMissions which would enable or disable missions for all the main quests like this one or yalahar gates etc...

**Review by gpedro — APPROVED** [review]

**Review by LukSrT — APPROVED** [review]

**Pull request merged**